Breeze.UI/src/app/wallet/advanced/advanced.service.ts

```diff
@@ -0,0 +1,51 @@
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs/Observable';
+import { HttpClient, HttpHeaders } from '@angular/common/http'; 
+import 'rxjs/add/observable/empty';
+
+import { GlobalService } from '../../shared/services/global.service';
+import { NavigationService, Page } from '../../shared/services/navigation.service';
+
+class dateRequest {
+    constructor(public date: Date){}
+}
+
+@Injectable()
+export class AdvancedService {
+    private urlPrefix = '';
+    private readonly walletName;
+    private readonly accountName = 'account 0';
+    private headers = new HttpHeaders({'Content-Type': 'application/json'});
+
+    constructor(private httpClient: HttpClient, private globalService: GlobalService, navigationService: NavigationService) { 
+        this.walletName = this.globalService.getWalletName();
+
+        navigationService.pageSubject.subscribe(x => 
+            this.urlPrefix = `http://localhost:3722${Page.Bitcoin ? 0 : 1}/api/Wallet/`);
+    }
+
+    public getExtPubKey(): Observable<string> {
+        const url = `${this.urlPrefix}extpubkey?WalletName=${this.walletName}&AccountName=${this.accountName}`;
+        return this.httpClient.get(url).map(x => x.toString());
+    }
+
+    public generateAddresses(count: number): Observable<string[]> {
+        const url = `${this.urlPrefix}unusedaddresses?WalletName=${this.walletName}&AccountName=${this.accountName}&Count=${count}`;
+        return this.httpClient.get(url).map(x => this.processAddresses(x));
+    }
+
+    public resyncFromDate(date: Date): Observable<any> {
+        date = new Date(date.getFullYear(), date.getMonth(), date.getDate()); //<- Strip any time values
```

Comment:
Just a question, can we get a null or undefined date passed in? If yes, we should handle it

Reply:
Don't think so, but will guard against it

Reply:
...actually no, we can't get null - the date is explicitly created by the component.